### DISCUSSION TITLE
Github Actions API list workflow runs purpose of parameter created

### DISCUSSION THREAD

**Discussion by teleivo** [community/community#26782] [Actions]
<p>Hi :smile:</p>
<p>the docs for listing runs of a Github action workflow specify a parameter <code>created</code> but do not say what it’s for. I only know its a string :grin:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/rest/reference/actions#list-workflow-runs">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/4661144/181099881-329d07c2-acb6-493b-a687-802db9e27ee1.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs" target="_blank" rel="noopener nofollow ugc">Actions - GitHub Docs</a></h3>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I also tried to look through the types of octokit/openapi spec <a href="https://raw.githubusercontent.com/octokit/openapi-types.ts/main/packages/openapi-types/types.d.ts" rel="noopener nofollow ugc">https://raw.githubusercontent.com/octokit/openapi-types.ts/main/packages/openapi-types/types.d.ts</a> which also just tells me its a string. I assume the docs are generated from the spec, or at least parts of it.</p>
<p>Would be interested to know what I can do with this parameter.<br>
Thank you for any hints :smile:</p>

**Comment by teleivo** [comment, marked as answer]
<p>You can filter runs by its <code>created_at</code> property.</p>
<p>See</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-09-02-github-actions-filter-workflow-runs-by-created-date/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-09-02-github-actions-filter-workflow-runs-by-created-date/" target="_blank" rel="noopener nofollow ugc">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/03/blog-card.png?fit=5001%2C2626" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2021-09-02-github-actions-filter-workflow-runs-by-created-date/" target="_blank" rel="noopener nofollow ugc">GitHub Actions: Filter workflow runs by created date | GitHub Changelog</a></h3>

  <p>GitHub Actions: Filter workflow runs by created date</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Interestingly, the actions API pagination only returns 10 pages even if it tells you in its link header that there are more than 10 pages. This is intentional as per GitHub support. Though not documented :yum: So if you want runs further in the past than the last 1000 (query param <code>per_page=100</code> and <code>page=10</code>) you can use the <code>created</code> param :grinning_face_with_smiling_eyes:</p>

**Reply by umassahu** [reply]
Can you please help me with an example as this is get API , when I passed created_at or created somehow its not working
GET : https://api.github.com/repos/XXX/XXX/actions/runs?created_date>2022-09-01

**Reply by teleivo** [reply]
This is an example request

`curl https://api.github.com/repos/USERNAME/REPOSITORY/actions/workflows/WORKFLOWID/runs?created=2022-09-21`

Fill in username, repository and workflow id (instead of my all caps placeholders) 😄 

You can find the workflow id via https://docs.github.com/en/rest/actions/workflows#list-repository-workflows.

It looks like you are trying to filter for created date greater than a certain date. I am not sure this is possible.
